src/Runtime/Distributions/DiscreteChar.cs

```diff
@@ -530,13 +542,24 @@ public void SetToProduct(DiscreteChar distribution1, DiscreteChar distribution2)
             }
 
             var builder = StorageBuilder.Create();
+
             foreach (var pair in CharRangePair.IntersectRanges(distribution1, distribution2))
             {
                 var probProduct = pair.Probability1 * pair.Probability2;
                 builder.AddRange(new CharRange(pair.StartInclusive, pair.EndExclusive, probProduct));
             }
 
-            this.Data = builder.GetResult();
+            double? logProbabilityOverride = null;
+            if (distribution1.Data.HasLogProbabilityOverride && distribution2.Data.IsBroad)
+            {
+                logProbabilityOverride = distribution1.Ranges.First().Probability.LogValue;
+            }
+            else if (distribution2.Data.HasLogProbabilityOverride && distribution1.Data.IsBroad)
+            {
+                logProbabilityOverride = distribution2.Ranges.First().Probability.LogValue;
+            }
+
```

Comment:
Should we have a check in case both have log probability overrides?  An exception thrown?

Reply:
I have put in a check